pkg/apis/camel/v1/maven_types.go

```diff
@@ -96,4 +96,9 @@ type Server struct {
 	Configuration Properties `xml:"configuration,omitempty" json:"configuration,omitempty"`
 }
 
-type Properties map[string]string
```

Comment:
This change here looks to be breaking compatibility in the API. In order to be performed we should either have a backward compatible field or deprecate the previous one and make it coexist for a couple of versions with the newer.

Reply:
@squakez I refactored the changes to only use the new plugin properties structure in Maven plugin execution configuration. The other occurrences of properties (e.g. in server configuration) now remain unchanged.

Reply:
I wonder if we could reach the same level of compatibility having `type Properties map[string]interface{}`. In this way, we don't need to include a new type and we neither have to deprecate the old one.

Reply:
Yeah, I have tried this one, too. It fails in client generator as it is not allowed to use generic interface{} at this point